src/main/java/org/apache/flink/benchmark/CheckpointingTimeBenchmark.java

```diff
@@ -285,7 +282,7 @@ protected int getNumberOfSlotsPerTaskManager() {
     public static class SlowDiscardSink<T> implements SinkFunction<T> {
         @Override
         public void invoke(T value, Context context) throws Exception {
-            Thread.sleep(1);
+            Thread.sleep(0, 500_000);
```

Comment:
This change shouldn't have any effect (take a look at the `Thead.sleep(millis, nanos)` implementation.

Has this change really speed up anything?

Reply:
That's stupid of me :facepalm:

Reply:
Decreasing the time of iteration works and the benchmark takes shorter, but for the cost of less invocations in an iteration. :( I will spend some more time on trying to understand and stabilise it further.